### DISCUSSION TITLE
Problem with triggering off INA226 value near startup

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#15276] [Q&A] [answered]
Still working on figuring out exactly what's happening, so haven't started an issue yet, but this doesn't occur on an identical system running 10.0.0.3, but does occur on 11.0.0.3

I have an INA226, a seven-segment display and two relays. If I have a rule that triggers immediately on startup like `on ina226-1#voltage>20.5 do power2 1 endon` (and INA is actually reading > 20.5v) the system bootloops on `restart 1` or a normal powerup until the failsafe happens that turns off the rules. If the rule doesn't trigger immediately, the system boots normally.

However, I have another ruleset that triggers off INA226 voltage and current values to send data to the display, and that does not crash the system on boot.

```
Rule1
On system#boot do backlog power3 1;ruletimer3 10 endon 
On ina226-1#Voltage!=var1 do var1 %value% endon 
On ina226-1#Current!=var2 do var2 %value% endon 
On rules#timer=3 do backlog displaytextnc %var1%; ruletimer2 2 endon 
On rules#timer=2 do backlog displaytextnc %var2%; ruletimer3 2 endon
```
Ruleset 1 works normally

```
Rule2
On ina226-1#voltage>20.5 do power2 1 endon
```
Ruleset 2 crashes the system

Both relays are on safe GPIOs (12 and 13). INA226 is on safe GPIOs (4 and 5).

If I change Rule2 to something that doesn't trigger on startup, like `On ina226-1#voltage>23.5 do power2 1 endon` there's no issue.

Am I missing something basic? I don't think so since this all worked on 10.0.0.3.

**Comment by pkkrusty** [comment, marked as answer]
I swapped the TM1637 from RX/TX gpios (so I could capture serial log data) to GPIO0 and GPIO2 and the problem has not reoccurred.  Not sure why that would have fixed the problem...

**Reply by pkkrusty** [reply]
Now thinking I may have been over-powering the board. I had 5.05v directly to the 5v pin, and the system wasn't booting. When I was debugging via USB, I was using USB power, which is quite a bit lower once it reaches the 5v rail (4.6v). Dropped the voltage down to 4.92 seems to have helped.

**Reply by sfromis** [reply]
What board is that? Usual development boards (with USB) has a voltage regulator stepping down the about 5V input to the 3.3V needed by the ESP chip. If providing near 5V to the ESP chip, high risk of troubles.

**Reply by barbudor** [reply]
Was it the pin swap or powering through USB that solved the problem ?
ie does it still crash when the TM is on 0/2 and still powered by your external PSU ?

5.05V is almost no difference to 5V. Most 5V parts would accept +5% or +10%.
I don't know the board you are using but if it happens to have a true AM1117 regulator, it can support up to 15V input.

If you have to time to investigate more, I would set it back in the configuration that crash and try to do replace the POWER command un ruleset 2 by a dummy command like setting an unused variable. If this does not crash, I would then try with Power2 1 but without the Relay 2 connected.
It may be possible that you power the relay early at the same time the ESP is trying to connect to Wifi which is when its power consumption is the maximum. That could make your power circuitry fails somehow.

**Reply by pkkrusty** [reply]
I've been powering my Wemos Mini D1 knockoffs with 5v directly to the 5v pin lately since powering them with USB tends to drop my 5v down too low to reliably activate mechanical relays (probably due to crappy regulator). 

I had to pin swap to deal get serial log data.  However, these 'relays' are MOSFET PWM boards, so the current draw is even less than the 50mA that a relay would take (I think). I did disconnect them to rule out funny business, but it made no difference.

It's running fine with the TM1637 on GPIO0/2 and powered via the external PSU at 4.92v

Just defined the TX/RX as CLK/DIO respectively, and it crashed in the same manner.  Seriallog was set to 0. Seems crazy that the display mapping to TX/RX would crash on a non-related trigger.

**Reply by pkkrusty** [reply]
ok, here's the serial log output on a crash cycle...
```
 ets Jan  8 2013,rst cause:4, boot mode:(3,7)                                   
                                                                                
wdt reset                                                                       
load 0x4010f000, len 3584, room 16                                              
tail 0                                                                          
chksum 0xb0                                                                     
csum 0xb0                                                                       
v3969889e                                                                       
~ld                                                                             
                                                                                
00:00:00.002 HDW: ESP8266EX                                                     
00:00:00.007 UFS: FlashFS mounted with 1992 kB free                             
00:00:00.051 CFG: Loaded from File, Count 294                                   
00:00:00.054 QPC: Count 1                                                       
00:00:00.061 DSP: TM1637 with 4 digits (type 0)                                 
                                                                                
 ets Jan  8 2013,rst cause:4, boot mode:(3,7)                                   
                                                                                
wdt reset                                                                       
load 0x4010f000, len 3584, room 16                                              
tail 0                                                                          
chksum 0xb0                                                                     
csum 0xb0                                                                       
v3969889e                                                                       
~ld                                                                             
                                                                                
00:00:00.001 HDW: ESP8266EX                                                     
00:00:00.007 UFS: FlashFS mounted with 1992 kB free                             
00:00:00.050 CFG: Loaded from File, Count 294                                   
00:00:00.053 FRC: Some settings have been reset (2)                             
00:00:00.060 DSP: TM1637 with 4 digits (type 0)                                 
00:00:00.062 SRC: Restart                                                       
                                                                                
 ets Jan  8 2013,rst cause:4, boot mode:(3,7)                                   
                                                                                
wdt reset                                                                       
load 0x4010f000, len 3584, room 16                                              
tail 0                                                                          
chksum 0xb0                                                                     
csum 0xb0                                                                       
v3969889e                                                                       
~ld                                                                             
                                                                                
00:00:00.001 HDW: ESP8266EX                                                     
00:00:00.006 UFS: FlashFS mounted with 1992 kB free                             
00:00:00.050 CFG: Loaded from File, Count 294                                   
00:00:00.053 FRC: Some settings have been reset (3)                             
00:00:00.059 DSP: TM1637 with 4 digits (type 0)                                 
00:00:00.062 SRC: Restart                                                       
                                                                                
 ets Jan  8 2013,rst cause:4, boot mode:(3,7)                                   
                                                                                
wdt reset                                                                       
load 0x4010f000, len 3584, room 16                                              
tail 0                                                                          
chksum 0xb0                                                                     
csum 0xb0                                                                       
v3969889e                                                                       
~ld      
```
Looks like it's the display doing it. Or is it the INA226 which initializes directly after the display...
```
00:00:00.001 HDW: ESP8266EX                                                     
00:00:00.008 UFS: FlashFS mounted with 1992 kB free                             
00:00:00.054 CFG: Loaded from File, Count 298                                   
00:00:00.057 FRC: Some settings have been reset (4)                             
00:00:00.063 DSP: TM1637 with 4 digits (type 0)                                 
00:00:00.066 SRC: Restart                                                       
00:00:00.069 Project pkkrusty - 24v Charger 2 Version 11.0.0.3(tasmota)-2_7_4_9)
00:00:00.072 I2C: INA226 found at 0x40                                          
00:00:00.256 CFG: Saved to flash at 3FB, Count 299, Bytes 4096     
```

**Reply by pkkrusty** [reply]
Hmm, now I'm thinking it's crashing on any active ruleset that's too long. I wonder if my flash memory is bad.

```
Rule3 
On time#minute=1 do power1 1 endon
On time#minute=2 do power2 1 endon
On time#minute=3 do power3 1 endon
On time#minute=4 do power1 1 endon
On time#minute=5 do power2 1 endon
On time#minute=6 do power3 1 endon
On time#minute=7 do power1 1 endon
On time#minute=8 do power1 1 endon
On time#minute=9 do power1 1 endon
On time#minute=10 do power1 1 endon
On time#minute=11 do power1 1 endon
On time#minute=12 do power1 1 endon
```
crashes, but
```
Rule3 
On time#minute=1 do power1 1 endon
```
does not.

will swap dev boards tomorrow and see if I can reproduce the problem.

**Reply by pkkrusty** [reply]
Swapped boards, but same problem occurs. Leaning towards a power issue, since I can't reproduce the problem powering the dev board via USB.
